### PR TITLE
new: l3 ping revamp and refreshID

### PR DESCRIPTION
**Pull request by sibicramesh**
https://github.com/aporeto-inc/aporeto-devs/issues/80
Usage https://github.com/aporeto-inc/enforcerd/pull/1693

### CODE REVIEW

**Comment by sibicramesh** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "l3-ping",
    "component": "enforcerd",
    "pr-id": "1693",
    "commit-sha": "7a688574cf0053bf602f5a09a1bc2d59c3cb8eba",
    "pipeline": "master"
  },
  {
    "project": "l3-ping",
    "component": "gaia",
    "pr-id": "481",
    "commit-sha": "6e60b61f1536d69338fe9de28deb24b01a9563c2",
    "pipeline": "master"
  },
  {
    "project": "l3-ping",
    "component": "backend",
    "pr-id": "735",
    "commit-sha": "8ffc5d1ec1c242dba486e8753ccbd290a8df0db5",
    "pipeline": "master"
  }
]
```

**Comment by sibicramesh** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "l3-ping",
    "component": "enforcerd",
    "pr-id": "1693",
    "commit-sha": "87dd29a77b2661f39dbbe382c6e7497282c53df4",
    "pipeline": "master"
  },
  {
    "project": "l3-ping",
    "component": "gaia",
    "pr-id": "481",
    "commit-sha": "8c0e370841d79577b00cec384b061c6d7ec13246",
    "pipeline": "master"
  },
  {
    "project": "l3-ping",
    "component": "backend",
    "pr-id": "735",
    "commit-sha": "a2fee6cbc1ec163e65e4d87494c8a2c09305778f",
    "pipeline": "master"
  }
]
```

**Comment by sibicramesh** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "l3-ping",
    "component": "enforcerd",
    "pr-id": "1693",
    "commit-sha": "ee06fbf9cbf4c33c4ac6029b29c0e561dca0a7e7",
    "pipeline": "master"
  },
  {
    "project": "l3-ping",
    "component": "gaia",
    "pr-id": "481",
    "commit-sha": "8c0e370841d79577b00cec384b061c6d7ec13246",
    "pipeline": "master"
  },
  {
    "project": "l3-ping",
    "component": "backend",
    "pr-id": "735",
    "commit-sha": "a2fee6cbc1ec163e65e4d87494c8a2c09305778f",
    "pipeline": "master"
  }
]
```

**Comment by sibicramesh** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "l3-ping",
    "component": "enforcerd",
    "pr-id": "1693",
    "commit-sha": "ee06fbf9cbf4c33c4ac6029b29c0e561dca0a7e7",
    "pipeline": "master"
  },
  {
    "project": "l3-ping",
    "component": "gaia",
    "pr-id": "481",
    "commit-sha": "8c0e370841d79577b00cec384b061c6d7ec13246",
    "pipeline": "master"
  },
  {
    "project": "l3-ping",
    "component": "backend",
    "pr-id": "735",
    "commit-sha": "23767630c0956c1c32566679cb22ff2b23cc6c9a",
    "pipeline": "master"
  }
]
```

**Comment by sibicramesh** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "l3-ping",
    "component": "enforcerd",
    "pr-id": "1693",
    "commit-sha": "5f6b27d65e1826de85206bbb1bb151b65c462578",
    "pipeline": "master"
  },
  {
    "project": "l3-ping",
    "component": "gaia",
    "pr-id": "481",
    "commit-sha": "8c0e370841d79577b00cec384b061c6d7ec13246",
    "pipeline": "master"
  },
  {
    "project": "l3-ping",
    "component": "backend",
    "pr-id": "735",
    "commit-sha": "23767630c0956c1c32566679cb22ff2b23cc6c9a",
    "pipeline": "master"
  }
]
```

**Comment by sibicramesh** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "l3-ping",
    "component": "enforcerd",
    "pr-id": "1693",
    "commit-sha": "c4da4c0154834dfc0abb51408798ea6d84fff0cb",
    "pipeline": "master"
  },
  {
    "project": "l3-ping",
    "component": "gaia",
    "pr-id": "481",
    "commit-sha": "8c0e370841d79577b00cec384b061c6d7ec13246",
    "pipeline": "master"
  },
  {
    "project": "l3-ping",
    "component": "backend",
    "pr-id": "735",
    "commit-sha": "23767630c0956c1c32566679cb22ff2b23cc6c9a",
    "pipeline": "master"
  }
]
```

**Comment by sibicramesh** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "l3-ping",
    "component": "enforcerd",
    "pr-id": "1693",
    "commit-sha": "1683c1d86fd0d99dcc29e4cd9b375cf75084b886",
    "pipeline": "master"
  },
  {
    "project": "l3-ping",
    "component": "gaia",
    "pr-id": "481",
    "commit-sha": "8c0e370841d79577b00cec384b061c6d7ec13246",
    "pipeline": "master"
  },
  {
    "project": "l3-ping",
    "component": "backend",
    "pr-id": "735",
    "commit-sha": "23767630c0956c1c32566679cb22ff2b23cc6c9a",
    "pipeline": "master"
  }
]
```

**Comment by sibicramesh** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "l3-ping",
    "component": "enforcerd",
    "pr-id": "1693",
    "commit-sha": "47dade69dca0f4750cad5a703e0effe0c4d45fec",
    "pipeline": "master"
  },
  {
    "project": "l3-ping",
    "component": "gaia",
    "pr-id": "481",
    "commit-sha": "8c0e370841d79577b00cec384b061c6d7ec13246",
    "pipeline": "master"
  },
  {
    "project": "l3-ping",
    "component": "backend",
    "pr-id": "735",
    "commit-sha": "23767630c0956c1c32566679cb22ff2b23cc6c9a",
    "pipeline": "master"
  }
]
```

**Comment by sibicramesh** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "l3-ping",
    "component": "enforcerd",
    "pr-id": "1693",
    "commit-sha": "fd3914aa3ed1eca66bf0ff9deaa9b846bae2ddd7",
    "pipeline": "master"
  },
  {
    "project": "l3-ping",
    "component": "gaia",
    "pr-id": "481",
    "commit-sha": "8c0e370841d79577b00cec384b061c6d7ec13246",
    "pipeline": "master"
  },
  {
    "project": "l3-ping",
    "component": "backend",
    "pr-id": "735",
    "commit-sha": "23767630c0956c1c32566679cb22ff2b23cc6c9a",
    "pipeline": "master"
  }
]
```

**Comment by sibicramesh** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "l3-ping",
    "component": "enforcerd",
    "pr-id": "1693",
    "commit-sha": "80157f7954d7d2b95909c4761aa03bea058bb641",
    "pipeline": "master"
  },
  {
    "project": "l3-ping",
    "component": "gaia",
    "pr-id": "481",
    "commit-sha": "8c0e370841d79577b00cec384b061c6d7ec13246",
    "pipeline": "master"
  },
  {
    "project": "l3-ping",
    "component": "backend",
    "pr-id": "735",
    "commit-sha": "23767630c0956c1c32566679cb22ff2b23cc6c9a",
    "pipeline": "master"
  }
]
```

**Comment by sibicramesh** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "l3-ping",
    "component": "enforcerd",
    "pr-id": "1693",
    "commit-sha": "1bf2290ff226e6bb24c35a49018116b6f788d69c",
    "pipeline": "master"
  },
  {
    "project": "l3-ping",
    "component": "gaia",
    "pr-id": "481",
    "commit-sha": "9c19bb12027dad05d8783416813c10353cd7ca72",
    "pipeline": "master"
  },
  {
    "project": "l3-ping",
    "component": "backend",
    "pr-id": "735",
    "commit-sha": "9fb67b32c315a01d600d7830be7933cd2cb042ef",
    "pipeline": "master"
  }
]
```

**Comment by sibicramesh** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "l3-ping",
    "component": "enforcerd",
    "pr-id": "1693",
    "commit-sha": "1bf2290ff226e6bb24c35a49018116b6f788d69c",
    "pipeline": "master"
  },
  {
    "project": "l3-ping",
    "component": "gaia",
    "pr-id": "481",
    "commit-sha": "9c19bb12027dad05d8783416813c10353cd7ca72",
    "pipeline": "master"
  },
  {
    "project": "l3-ping",
    "component": "backend",
    "pr-id": "735",
    "commit-sha": "8788276b33f9e712219054165b7cc5ce6617664d",
    "pipeline": "master"
  }
]
```

File: specs/pingreport.spec [hunk 1]
```diff
@@ -85,22 +113,28 @@ attributes:
...
+    description: Current stage when this report has been generated.
     type: string
     exposed: true
 
   - name: timestamp
     description: Date of the report.
     type: time
     exposed: true
+
+  - name: txController
```
Comment (primalmotion) [inline specs/pingreport.spec:127]: Should be TXController

File: specs/pingreport.spec [hunk 2]
```diff
@@ -72,10 +84,26 @@ attributes:
     exposed: true
 
   - name: request
-    description: Request represents the request number.
+    description: Request represents the current request.
     type: integer
     exposed: true
 
+  - name: rxFourTuple
```
Comment (primalmotion) [inline specs/pingreport.spec:91]: Should be RX

**Comment by sibicramesh** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "l3-ping",
    "component": "enforcerd",
    "pr-id": "1693",
    "commit-sha": "edee0ecdf472033f099bf51bd5429996e3ffe3f6",
    "pipeline": "master"
  },
  {
    "project": "l3-ping",
    "component": "gaia",
    "pr-id": "481",
    "commit-sha": "d826c235dd33378655625d92ca85efd4b65a673d",
    "pipeline": "master"
  },
  {
    "project": "l3-ping",
    "component": "backend",
    "pr-id": "735",
    "commit-sha": "69978541a76bb569800b6f50dbccb020005f1a6a",
    "pipeline": "master"
  }
]
```

**Comment by sibicramesh** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "l3-ping",
    "component": "enforcerd",
    "pr-id": "1693",
    "commit-sha": "edee0ecdf472033f099bf51bd5429996e3ffe3f6",
    "pipeline": "master"
  },
  {
    "project": "l3-ping",
    "component": "gaia",
    "pr-id": "481",
    "commit-sha": "567c46eb97ebfbddb82206913b30aa968ec6aa75",
    "pipeline": "master"
  },
  {
    "project": "l3-ping",
    "component": "backend",
    "pr-id": "735",
    "commit-sha": "c6b9cf7d09a665790732d9d9dc2c3014389f8537",
    "pipeline": "master"
  }
]
```

**Comment by sibicramesh** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "l3-ping",
    "component": "enforcerd",
    "pr-id": "1693",
    "commit-sha": "ede3560b749c787dc169a10eb8438bf1ea261cd0",
    "pipeline": "master"
  },
  {
    "project": "l3-ping",
    "component": "gaia",
    "pr-id": "481",
    "commit-sha": "567c46eb97ebfbddb82206913b30aa968ec6aa75",
    "pipeline": "master"
  },
  {
    "project": "l3-ping",
    "component": "backend",
    "pr-id": "735",
    "commit-sha": "c6b9cf7d09a665790732d9d9dc2c3014389f8537",
    "pipeline": "master"
  }
]
```

**Pull request merged**